src/autolabel/confidence.py

```diff
@@ -149,6 +168,18 @@ def calculate(self, model_generation: LLMAnnotation, **kwargs) -> float:
             keys=keys,
             **kwargs,
         )
+        model_generation.confidence_score = confidence
+
+        # On cache miss, update cache
```

Comment:
since caching is only done for the case where the llm does not return logprobs - lets move the code in that if condition. here in the case we have the llm return logprobs then we will have cache_miss variable not defined error